lib/core/src/Cardano/Wallet.hs

```diff
@@ -1968,43 +1969,39 @@ selectAssets ctx pp params transform = do
         txWithdrawal = params ^. (#txContext . #txWithdrawal)
 
 signTransaction
-    :: forall ctx s k.
-        ( HasTransactionLayer k ctx
-        , HasDBLayer IO s k ctx
-        , HasNetworkLayer IO ctx
-        , HardDerivation k
-        , WalletKey k
-        , IsOwned s k
-        , Bounded (Index (AddressIndexDerivationType k) 'AddressK)
-        )
-    => ctx
-    -> WalletId
-    -> Passphrase "raw"
-    -> SealedTx
-    -> ExceptT ErrWitnessTx IO SealedTx
-signTransaction ctx wid pwd tx = db & \DBLayer{..} -> do
-    era <- liftIO $ currentNodeEra nl
-    withRootKey @_ @s ctx wid pwd ErrWitnessTxWithRootKey $ \rootK scheme -> do
-        cp <- mapExceptT atomically
-            $ withExceptT ErrWitnessTxNoSuchWallet
-            $ withNoSuchWallet wid
-            $ readCheckpoint wid
-        let pwdP = preparePassphrase scheme pwd
-        let rewardAcnt = (getRawKey $ deriveRewardAccount @k pwdP rootK, pwdP)
-        let addressResolver = mkAddressResolver cp (rootK, pwdP)
-        let inputResolver = mkInputResolver cp
-        pure $ addVkWitnesses tl era rewardAcnt addressResolver inputResolver tx
-  where
-    db = ctx ^. dbLayer @IO @s @k
-    tl = ctx ^. transactionLayer @k
-    nl = ctx ^. networkLayer
-
-    mkInputResolver cp i = do
-        TxOut addr _ <- UTxO.lookup i (totalUTxO mempty cp)
-        pure addr
-
-    mkAddressResolver cp (rootK, pwdP) =
-        isOwned (getState cp) (rootK, pwdP)
+  :: forall k
+   . ( WalletKey k
+     , HardDerivation k
+     , Bounded (Index (AddressIndexDerivationType k) 'AddressK)
+     )
+  => TransactionLayer k SealedTx
+  -- ^ The way to interact with the wallet backend
+  -> Cardano.AnyCardanoEra
+  -- ^ The era to operate in
+  -> (Address -> Maybe (k 'AddressK XPrv, Passphrase "encryption"))
+  -- ^ The wallets address-key lookup function
+  -> (k 'RootK XPrv, Passphrase "encryption")
+  -- ^ The root key of the wallet
+  -> UTxO
+  -- ^ The total UTxO set of the wallet (i.e. if pending transactions all
+  -- applied).
+  -> SealedTx
+  -- ^ The transaction to sign
+  -> SealedTx
+  -- ^ The original transaction, with additional signatures added where
+  -- necessary
+signTransaction tl era keyLookup (rootKey, rootPwd) utxo =
+    let
+        rewardAcnt :: (XPrv, Passphrase "encryption")
```

Comment:
what about exposing this function? It is used also in tests. Maybe it would be good to have code reuse